extensions.html

```diff
@@ -445,16 +507,166 @@ <h2>
+          </p>
+          <ol>
+            <li>If [=this=].{{Gamepad/[[touchEvents]]}} not `null` and not empty,
+              return [=this=].{{Gamepad/[[touchEvents]]}}.
+            </li>
+            <li>Otherwise return `null`.
+            </li>
+          </ol>
+
         </dd>
       </dl>
+      <section>
+        <h3>
+          Receiving inputs
+        </h3>
+        <p>
+          This section supplements the <a href="https://w3c.github.io/gamepad/#receiving-inputs">
+          Receiving inputs</a> section of the main 
+          <a href="https://w3c.github.io/gamepad/">Gamepad specification</a>.
+        </p>
+        <p>
+          In addtion to the steps defined in the main 
+          <a href="https://w3c.github.io/gamepad/">Gamepad specification</a>.
+          When the user agent modifies the list of |active touch points| for a 
+          touch surface by adding touch points, removing touch points, 
+          or updating the values of existing touch points,
+          <dfn data-lt="updating touchEvents">update touchEvents</dfn>
+          by running the following steps:
+        </p>
+        <ol>
+          <li>Let |surfaceId:unsigned long| be 0.
+          </li>
+          <li>Remove any existing events from {{Gamepad}}.{{Gamepad/[[touchEvents]]}}.
+          </li>
+          <li>Repeat the following steps for each [=touch surface=] on |gamepad| in 
+            [=touch surface enumeration order=]:
+            <ol>
+              <li>Let |surfaceDimensions| be an `null`.
+              </li>
+              <li>If the touch surface exposes maximum surface dimensions in device units:
+                <ol>
+                  <li>Set |touchEvent|.{{GamepadTouch/surfaceDimensions}}[0] to the maximum
+                    X dimension on the [=touch surface=] in device units.
+                  </li>
+                  <li>Set |touchEvent|.{{GamepadTouch/surfaceDimensions}}[1] to the maximum
+                    Y dimension on the [=touch surface=] in device units.
+                  </li>
+                </ol>
+              </li>
+              <li>Repeat the following steps for each 
+                [=active touch point=]
+                reported by the |gamepad| for the current [=touch surface=].
+                <ol>
+                  <li>
+                    Let |nextTouchId:unsigned long| be the next available touchId for the |gamepad|.
+                  </li>
+                  <li>Let |touchEvent:GamepadTouch| be a {{GamepadTouch}}.
+                  </li>
+                  <li>Set |touchEvent|.{{GamepadTouch/surfaceId}} to be
+                    |surfaceId|.
+                  </li>
+                  <li>If the touch data is part of an existing 
+                    [=active touch point=]
+                    tracked by the user agent:
+                    <ol>
+                    <li>Set |touchEvent|.{{GamepadTouch/touchId}} to the touchId of the [=active touch point=].
```

Comment:
Please use [HTML Tidy](https://www.html-tidy.org/) to update the formatting in this file.

This command will fix formatting errors in extensions.html, overwriting the input file:

`tidy -config tidyconfig.txt -m extensions.html`